electron/main.ts

```diff
@@ -262,20 +262,53 @@ ipcMain.on('quit', async () => handleExit())
 /* const storage: Storage = mainWindow.localStorage
 const lang = storage.getItem('language') */
 
-ipcMain.handle('getGameInfo', async (event, game) => {
+const fetchGraphQl = async (namespace: string, game:string) => {
+  const graphql = JSON.stringify({
+    query: `{Catalog{catalogOffers( namespace:"${namespace}"){elements {productSlug}}}}`,
+    variables: {}
+  })
+  const result = await axios("https://www.epicgames.com/graphql", {
+    method: 'POST',
+    headers: { "Content-Type": "application/json" },
+    data: graphql
+  })
+  const res = result.data.data.Catalog.catalogOffers
+  const slug = res.elements.find((e: { productSlug: string }) => e.productSlug)
+  if (slug) {
+    console.log('SLUG',slug)
+    return slug.productSlug.replace(/(\/.*)/, '')
+  } else {
+    console.log('game',game)
+    return game
+  }
+}
+
+
+
+
+ipcMain.handle('getGameInfo', async (event, game, namespace: string | null) => {
   let lang = JSON.parse(readFileSync(heroicConfigPath, 'utf-8')).defaultSettings
     .language
   if (lang === 'pt') {
     lang = 'pt-BR'
   }
-  const epicUrl = `https://store-content.ak.epicgames.com/api/${lang}/content/products/${game}`
+
+  let epicUrl:string;
+  if (namespace) {
+    const fetch:string = await fetchGraphQl(namespace, game)
```

Comment:
Fetch is super ambiguous and could cause confusion upon reading.

Reply:
Good point. Also fetch is a javascript function. Try something like getSomething... since this is an action.